packages/core/src/storage/vectorStore/ChromaVectorStore.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AddParams,
+  ChromaClient,
+  ChromaClientParams,
+  Collection,
+  IncludeEnum,
+  QueryResponse,
+  Where,
+  WhereDocument,
+} from "chromadb";
+import { BaseNode, MetadataMode } from "../../Node";
+import {
+  VectorStore,
+  VectorStoreQuery,
+  VectorStoreQueryMode,
+  VectorStoreQueryResult,
+} from "./types";
+import { metadataDictToNode, nodeToMetadata } from "./utils";
+
+type ChromaDeleteOptions = {
+  where?: Where;
+  whereDocument?: WhereDocument;
+};
+
+type ChromaQueryOptions = {
+  whereDocument?: WhereDocument;
+};
+
+const DEFAULT_TEXT_KEY = "text";
+
+export class ChromaVectorStore implements VectorStore {
+  storesText: boolean = true;
+  flatMetadata: boolean = true;
+  textKey: string;
+  private chromaClient: ChromaClient;
+  private collection: Collection | null = null;
+  private collectionName: string;
+
+  constructor(init: {
+    collectionName: string;
+    textKey?: string;
+    chromaClientParams?: ChromaClientParams;
+  }) {
+    this.collection = null;
+    this.collectionName = init.collectionName;
+    this.chromaClient = new ChromaClient(init.chromaClientParams);
+    this.textKey = init.textKey ?? DEFAULT_TEXT_KEY;
+  }
+
+  client(): ChromaClient {
+    return this.chromaClient;
+  }
+
+  async getCollection(): Promise<Collection> {
+    if (!this.collection) {
+      const coll = await this.chromaClient.createCollection({
+        name: this.collectionName,
+      });
+      this.collection = coll;
+    }
+    return this.collection;
+  }
+
+  private getDataToInsert(nodes: BaseNode[]): AddParams {
+    const metadatas = nodes.map((node) =>
+      nodeToMetadata(node, true, this.textKey, this.flatMetadata),
+    );
+    return {
+      embeddings: nodes.map((node) => node.getEmbedding()),
+      ids: nodes.map((node) => node.id_),
+      metadatas,
+      documents: nodes.map((node) => node.getContent(MetadataMode.NONE)),
+    };
+  }
+
+  async add(nodes: BaseNode[]): Promise<string[]> {
+    if (!nodes || nodes.length === 0) {
+      return [];
+    }
+
+    const dataToInsert = this.getDataToInsert(nodes);
+    const collection = await this.getCollection();
+    await collection.add(dataToInsert);
+    return nodes.map((node) => node.id_);
+  }
+
+  async delete(
+    refDocId: string,
+    deleteOptions?: ChromaDeleteOptions,
+  ): Promise<void> {
+    const collection = await this.getCollection();
+    await collection.delete({
+      ids: [refDocId],
+      where: deleteOptions?.where,
+      whereDocument: deleteOptions?.whereDocument,
+    });
+    this.collection = null;
```

Comment:
is this really needed? it will force a re-init of `collection` (calling `this.chromaClient.createCollection`)